src/Commands/HydroponicsMultiDrushCommands.php

```diff
@@ -0,0 +1,229 @@
+          $process->on('exit', function ($code, $term) use ($loop, $queue, $process_pid, &$child, $timeout){
+            // Do we need this? YES, to remove timeout timer if exit before timeout
+            $loop->cancelTimer($timeout[$queue][$process_pid]);
+
+            $child[$queue][$process_pid]['end'] = \Drupal::time()->getCurrentTime();
+            $child[$queue][$process_pid]['exitcode'] = $code;
+            \Drupal::logger('hydroponics')->info("EXIT event: Queue @queue, process @pid, output @output, exit code @code, term @term, start @start, end @end, timeout @timeout", [
+              '@queue' => $queue,
+              '@pid' => $process_pid,
+              '@output' => is_null($child[$queue][$process_pid]['output']) ? "NULL" : $child[$queue][$process_pid]['output'],
+              '@code' => is_null($child[$queue][$process_pid]['exitcode']) ? "NULL" : $child[$queue][$process_pid]['exitcode'],
+              '@term' => is_null($term) ? "NULL" : $term,
+              '@start' => is_null($child[$queue][$process_pid]['start']) ? "NULL" : $child[$queue][$process_pid]['start'],
+              '@end' => is_null($child[$queue][$process_pid]['end']) ? "NULL" : $child[$queue][$process_pid]['end'],
+              '@timeout' => $child[$queue][$process_pid]['timeout'] ? "TRUE" : "FALSE"
+            ]);
+          });
+
+        }
+        else {
+          // not able to run child due to queue empty or max child reached
+          \Drupal::logger('hydroponics')->info("Queue empty OR max @max process running on queue @queue", [
+            '@max' => $max_child_for_queue,
+            '@queue' => $queue
+          ]);
+          // If no child running and queue empty then decrement idle timer
+          if (($running_child == 0) && ($items[$queue] == 0)) {
+            $idle[$queue] -= 1;
+          }
+        }
+      });
+    }
+
+    // idle check every $idle_timer_time s
+    $idle_timer = $loop->addPeriodicTimer($idle_timer_time, function ($timer) use ($loop, $timer_ping, &$done, &$idle) {
+      // Close main loop if all queues return 0 elements for at least N cycles
+      // ToDO also check or report if some child timeout
+      $all_idle = 1;
+      foreach($idle as $queue_idle) {
+        if ($queue_idle > 0) {
+          $all_idle = 0;
+        }
+      }
+      if ($all_idle === 1) {
+        $loop->cancelTimer($timer_ping);
+        foreach($done as $queue_timer) {
+          $loop->cancelTimer($queue_timer);
+        }
+        $queuerunner_pid = (int) \Drupal::state()->get('hydroponics.queurunner_last_pid', 0);
+        //set pid to negative to avoid lost of information in case of hang
+        $queuerunner_pid = $queuerunner_pid * (-1);
+        \Drupal::state()->set('hydroponics.queurunner_last_pid', $queuerunner_pid);
+        \Drupal::logger('hydroponics')->info("All queues are idle, closing timers");
+
+        $loop->cancelTimer($timer);
+        }
+      }
+    );
+
+    /* TODO recompile with PCNTL enabled
```

Comment:
This comments can be removed. Better to clean up here before merging

Reply:
done